EssentialsDiscord/src/main/resources/config.yml

```
@@ -0,0 +1,295 @@
+  # - {controllername}: The username of the user who muted the player
+  # - {controllerdisplayname}: The display name of the user who muted the player
+  # - {reason}: The reason the player was muted for
+  permanent-mute-reason: "{controllerdisplayname} permanently muted {displayname} with reason: {reason}"
+  # This is the message sent to discord when a player is unmuted in minecraft.
+  # The following placeholders can be used here:
+  # - {username}: The username of the player being unmuted
+  # - {displayname}: The display name of the player being unmuted
+  unmute: "{displayname} has been unmuted."
+  # This is the message sent to discord when a player joins the minecraft server.
+  # The following placeholders can be used here:
+  # - {username}: The name of the user joining
+  # - {displayname}: The display name of the user joining
+  # - {defaultmessage}: The full default join message used in game
+  # ... PlaceholderAPI placeholders are also supported here too!
+  join: ":exclamation: {displayname} has joined!"
+  # This is the message sent to discord when a player leaves the minecraft server.
+  # The following placeholders can be used here:
+  # - {username}: The name of the user leaving
+  # - {displayname}: The display name of the user leaving
+  # - {defaultmessage}: The full default leave message used in game
+  # ... PlaceholderAPI placeholders are also supported here too!
+  quit: ":exclamation: {displayname} has left!"
+  # This is the message sent to discord when a player dies.
+  # The following placeholders can be used here:
+  # - {username}: The name of the user who died
+  # - {displayname}: The display name of the user who died
+  # - {defaultmessage}: The full default death message used in game
+  # ... PlaceholderAPI placeholders are also supported here too!
+  death: ":skull: {displayname} has died!"
+  # This is the message sent to discord when a player becomes afk.
+  # The following placeholders can be used here:
+  # - {username}: The name of the user who became afk
+  # - {displayname}: The display name of the user who became afk
+  # ... PlaceholderAPI placeholders are also supported here too!
+  afk: ":person_walking: {displayname} is now AFK!"
+  # This is the message sent to discord when a player is no longer afk.
+  # The following placeholders can be used here:
+  # - {username}: The name of the user who is no longer afk
+  # - {displayname}: The display name of the user who is no longer afk
+  # ... PlaceholderAPI placeholders are also supported here too!
+  un-afk: ":keyboard: {displayname} is no longer AFK!"
+  # This is the message sent to discord when a player is kicked from the server.
+  # The following placeholders can be used here:
+  # - {username}: The name of the user who got kicked
+  # - {displayname}: The display name of the user who got kicked
+  # - {reason}: The reason the player was kicked
+  kick: "{displayname} was kicked with reason: {reason}"
```

```suggestion
  kick: "{controllerdisplayname} kicked {displayname} for {reason}."
```
Note: `controllerdisplayname` does not seem to be supported here right now (though normally shown in Essentials).

---

we use PlayerKickEvent (because we want to support more than just essx kicks) which doesn't have players as causes so we cannot support controllers